flexmeasures/data/services/timerange.py

```diff
@@ -0,0 +1,38 @@
+from __future__ import annotations
+
+from datetime import datetime
+
+from flexmeasures.utils import time_utils
+
+
+def get_timerange(sensor_ids: list[int]) -> tuple[datetime, datetime]:
+    """Get the start and end of the least recent and most recent event, respectively."""
+    from flexmeasures.data.models.time_series import TimedBelief
+
+    least_recent_query = (
+        TimedBelief.query.filter(TimedBelief.sensor_id.in_(sensor_ids))
+        .order_by(TimedBelief.event_start.asc())
+        .limit(1)
+    )
+    most_recent_query = (
+        TimedBelief.query.filter(TimedBelief.sensor_id.in_(sensor_ids))
+        .order_by(TimedBelief.event_start.desc())
+        .limit(1)
```

Comment:
Maybe, instead of doing a `ORDER BY` + `LIMIT`, I would use the functions `MAX(event_end)`, `MIN(event_start)`. 

[It turns out](https://stackoverflow.com/questions/426731/min-max-vs-order-by-and-limit) that for an indexed column It's not much of a difference, but for unindexed columns it can.